server/src/main/java/org/elasticsearch/action/admin/cluster/shards/ClusterSearchShardsGroup.java

```diff
@@ -84,4 +110,47 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.endArray();
         return builder;
     }
+
+    public static ClusterSearchShardsGroup fromXContent(XContentParser parser) throws IOException {
+        ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.currentToken(), parser::getTokenLocation);
+        parser.nextToken();
+        List<ShardRouting> routings = new ArrayList<>();
+        while (parser.currentToken() != XContentParser.Token.END_ARRAY) {
+            final ShardRouting routing = ShardRouting.fromXContent(parser);
+            routings.add(routing);
+            parser.nextToken();
+        }
+
+        ensureExpectedToken(XContentParser.Token.END_ARRAY, parser.currentToken(), parser::getTokenLocation);
+
+        if (!routings.isEmpty()) {
```

Comment:
We tend to use `routings.isEmpty() == false` since it's harder to typo

Reply:
Actually, I'd probably reverse the check so it was:

```java
if (routings.isEmpty()) {
    return new ClusterSearchShardsGroup();
}
... rest of code ...
```

Reply:
agreed